### PR TITLE
Support AVAS in Hamiltonian generation

### PR DESCRIPTION
**Pull request by max-radin**
## Description

This PR allows Hamiltonians to be generated for chemistry application instances that use AVAS to reduce the active space.

## Please verify that you have completed the following steps

- [x] I have self-reviewed my code.
- [x] I have included test cases validating introduced feature/fix.
- [x] I have updated documentation.

### CODE REVIEW

File: src/benchq/problem_ingestion/molecule_instance_generation.py [hunk 1]
```diff
@@ -117,12 +179,7 @@ def truncate_with_avas(
         mean_field_object, loc_type="pm"
     )  # default is loc_type ='pm' (Pipek-Mezey)
 
-    # Truncates to a specific active space that captures the essential chemistry
-    molecule, mean_field_object = avas_active_space(
-        mean_field_object, ao_list=ao_list, minao=minao
-    )
-
-    return mean_field_object
+    return avas_active_space(mean_field_object, ao_list=ao_list, minao=minao)
```
Comment (mstechly) [inline src/benchq/problem_ingestion/molecule_instance_generation.py:182]: Add a return type perhaps?

File: tests/benchq/problem_ingestion/test_molecule_instance_generation.py [hunk 2]
```diff
@@ -6,12 +6,17 @@
 )
 
 
+
+avas_hydrogen_chain_instance = generate_hydrogen_chain_instance(2)
```
Comment (mstechly) [inline tests/benchq/problem_ingestion/test_molecule_instance_generation.py:10]: Could you encapsulate it in:
```
def _generate_avas_hydrogen_chain_instance(n_hydrogens):
    avas_hydrogen_chain_instance = generate_hydrogen_chain_instance(n_hydrogens)
    avas_hydrogen_chain_instance.avas_atomic_orbitals = ["H 1s"]
    avas_hydrogen_chain_instance.avas_minao = "sto-3g"
    return avas_hydrogen_chain_instance
```
Or sth like that?
I got a bit confused by what's going on here.

**Review by mstechly — APPROVED** [review]

File: src/benchq/problem_ingestion/molecule_instance_generation.py [hunk 3]
```diff
@@ -48,49 +49,111 @@ class ChemistryApplicationInstance:
...
+            The fermionic Hamiltonian corresponding to the instance's active space. Note
+                that the active space will account for both AVAS and the
+                occupied_indices/active_indices attributes.
+        """
+        return self._get_molecular_data().get_molecular_hamiltonian(
             occupied_indices=self.occupied_indices,
             active_indices=self.active_indices,
         )
 
     def get_active_space_meanfield_object(self) -> scf.hf.SCF:
```
Comment (mkodrycka) [inline src/benchq/problem_ingestion/molecule_instance_generation.py:99]: Maybe we should rename method's signature so it indicates that we get active space accounting for AVAS? WDYT?

File: src/benchq/problem_ingestion/molecule_instance_generation.py [hunk 4]
```diff
@@ -48,49 +49,111 @@ class ChemistryApplicationInstance:
...
-        return truncate_with_avas(
-            self.get_molecular_data()._pyscf_data["scf"],
-            self.avas_atomic_orbitals,
-            self.avas_minao,
+        return self._run_pyscf()[1]
+
+    def _get_molecular_data(self):
+        """Given a PySCF meanfield object and molecule, return a PyscfMolecularData object.
+
+        Args:
```
Comment (mkodrycka) [inline src/benchq/problem_ingestion/molecule_instance_generation.py:119]: I think the list of args is redundant here.

**Review by mkodrycka — APPROVED** [review]
Thanks for adding this feature! It looks really good! :) 

I added a couple of small comments.

File: src/benchq/problem_ingestion/molecule_instance_generation.py (hunk 4 above)
Comment (max-radin) [reply src/benchq/problem_ingestion/molecule_instance_generation.py]: Good catch, fixed in a2216750a1c9042466a4c39aff99db5299e1afe9

File: src/benchq/problem_ingestion/molecule_instance_generation.py (hunk 1 above)
Comment (max-radin) [reply src/benchq/problem_ingestion/molecule_instance_generation.py:179]: Fixed in 549744d2e9d2e0ef8ba2353841fbe8eb69c229cd

File: tests/benchq/problem_ingestion/test_molecule_instance_generation.py (hunk 2 above)
Comment (max-radin) [reply tests/benchq/problem_ingestion/test_molecule_instance_generation.py]: Fixed in a2216750a1c9042466a4c39aff99db5299e1afe9. This is just creating one of the application instances to be used in the following test

Comment (mstechly) [reply tests/benchq/problem_ingestion/test_molecule_instance_generation.py]: I have figured it, but it took me half a minute or so ;)
Thanks!

File: src/benchq/problem_ingestion/molecule_instance_generation.py (hunk 3 above)
Comment (max-radin) [reply src/benchq/problem_ingestion/molecule_instance_generation.py:103]: Good question. I had left AVAS out of the name with the idea that the person who wants to get the active space Hamiltonian probably doesn't know about what methods a particular application instance uses to construct the active space (AVAS, freezing orbitals with certain indices, or in the future some other techniques).

I did update the docstring to make this more clear in 356fbaf99168864c11c. What do you think?

Comment (mkodrycka) [reply src/benchq/problem_ingestion/molecule_instance_generation.py:103]: I like it! Thanks.

**Pull request merged**